File: plugin-rust-agent/src/main/java/jetbrains/buildServer/rust/cargo/LoginArgumentsProvider.java
/*
 * Copyright 2000-2016 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * See LICENSE in the project root for license information.
 */

package jetbrains.buildServer.rust.cargo;

import jetbrains.buildServer.rust.ArgumentsProvider;
import jetbrains.buildServer.rust.CargoConstants;
import jetbrains.buildServer.util.StringUtil;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * Provides arguments to cargo login command.
 */
public class LoginArgumentsProvider implements ArgumentsProvider {

    @NotNull
    @Override
    public List<String> getArguments(@NotNull final Map<String, String> parameters) {
        final List<String> arguments = new ArrayList<String>();
        arguments.add(CargoConstants.COMMAND_LOGIN);

        final String hostValue = parameters.get(CargoConstants.PARAM_LOGIN_HOST);
        if (!StringUtil.isEmptyOrSpaces(hostValue)) {
            arguments.add("--host");
            arguments.add(hostValue.trim());
        }

        final String verbosityValue = parameters.get(CargoConstants.PARAM_VERBOSITY);
        if (!StringUtil.isEmptyOrSpaces(verbosityValue)) {
            arguments.add(verbosityValue.trim());
        }

        final String tokenValue = parameters.get(CargoConstants.PARAM_LOGIN_TOKEN);
        if (!StringUtil.isEmptyOrSpaces(tokenValue)) {
            arguments.add(tokenValue.trim());
        }

        return arguments;
    }
}

File: plugin-rust-agent/src/main/kotlin/jetbrains/buildServer/rust/cargo/LoginArgumentsProvider.kt


package jetbrains.buildServer.rust.cargo

import jetbrains.buildServer.agent.BuildRunnerContext
import jetbrains.buildServer.rust.ArgumentsProvider
import jetbrains.buildServer.rust.CargoConstants
import java.util.*

/**
 * Provides arguments to cargo login command.
 */
class LoginArgumentsProvider : ArgumentsProvider {

    override fun getArguments(runnerContext: BuildRunnerContext): List<String> {
        val parameters = runnerContext.runnerParameters
        val arguments = ArrayList<String>()
        arguments.add(CargoConstants.COMMAND_LOGIN)

        val hostValue = parameters[CargoConstants.PARAM_LOGIN_HOST]
        if (!hostValue.isNullOrBlank()) {
            arguments.add("--host")
            arguments.add(hostValue.trim())
        }

        addCommonArguments(parameters, arguments)

        val tokenValue = parameters[CargoConstants.PARAM_LOGIN_TOKEN_SECURE] ?: parameters[CargoConstants.PARAM_LOGIN_TOKEN]
        if (!tokenValue.isNullOrBlank()) {
            arguments.add(tokenValue.trim())
        }

        return arguments
    }
}